admiral/pkg/clusters/types_test.go

```diff
@@ -257,5 +262,184 @@ func TestHandleEventForGlobalTrafficPolicy(t *testing.T) {
+	registry.AdmiralCache.IdentityDependencyCache.Put("foo1", "bar1", "bar1")
+
+
+	handler.RemoteRegistry = registry
+
+	routingPolicyFoo := &v1.RoutingPolicy{
+		TypeMeta:   time2.TypeMeta{},
+		ObjectMeta: time2.ObjectMeta{
+			Labels: map[string]string{
+				"identity": "foo",
+				"admiral.io/env": "stage",
+			},
+		},
+		Spec:       model.RoutingPolicy{
+			Plugin:               "test",
+			Hosts:                []string{"e2e.testservice.mesh"},
+			Config: map[string]string{
+				"cachePrefix": "cache-v1",
+				"cachettlSec": "86400",
+				"routingServiceUrl": "e2e.test.routing.service.mesh",
+				"pathPrefix": "/sayhello,/v1/company/{id}/",
+			},
+		},
+		Status:     v1.RoutingPolicyStatus{},
+	}
+
+	routingPolicyFoo1 := routingPolicyFoo.DeepCopy()
+	routingPolicyFoo1.Labels[common.GetWorkloadIdentifier()] = "foo1"
+
+
+	testCases := []struct {
+		name 						string
+		routingPolicy				*v1.RoutingPolicy
+		expectedFilterCacheKey		string
+		valueExpected				bool
+
+	} {
+		{
+			name:                     "should have the right envoyFilter in filter cache",
+			routingPolicy:             routingPolicyFoo,
+			expectedFilterCacheKey:	  "barstage",
+			valueExpected:			  true,
+		},
+		{
+			name:                     "If dependent deployment does not exist, the filter should not be created",
+			routingPolicy:             routingPolicyFoo1,
+			expectedFilterCacheKey:	  "bar1stage",
+			valueExpected:			  false,
+		},
+		{
+			name:                     "If dependent deployment exists in a different cluster, the filter should not be created",
+			routingPolicy:             routingPolicyFoo,
+			expectedFilterCacheKey:	  "bar2stage",
+			valueExpected:			  false,
+		},
+
+	}
+
+	for _, c := range testCases {
+		t.Run(c.name, func(t *testing.T) {
+			handler.Added(c.routingPolicy)
+			if c.valueExpected {
+				filterCacheValue := registry.AdmiralCache.RoutingPolicyFilterCache.Get(c.expectedFilterCacheKey)
+				assert.NotNil(t, filterCacheValue)
+				selectorLabelsSha, err := common.GetSha1("bar"+common.GetRoutingPolicyEnv(c.routingPolicy))
+				if err != nil {
+					t.Error("Error ocurred while computing workload Labels sha1")
+				}
+				envoyFilterName := fmt.Sprintf("%s-dynamicrouting-%s-%s", strings.ToLower(c.routingPolicy.Spec.Plugin), selectorLabelsSha, "1.10")
+				filterMap := filterCacheValue[remoteController.ClusterID]
+				assert.NotNil(t, filterMap)
+				assert.NotNil(t, filterMap[envoyFilterName])
+
+				// once the routing policy is deleted, the corresponding filter should also be deleted
+				handler.Deleted(c.routingPolicy)
+				assert.Nil(t, registry.AdmiralCache.RoutingPolicyFilterCache.Get(c.expectedFilterCacheKey))
+			} else {
+				assert.Nil(t, registry.AdmiralCache.RoutingPolicyFilterCache.Get(c.expectedFilterCacheKey))
+			}
+
+		})
+	}
+
+	// Test for multiple filters
+	registry.AdmiralCache.IdentityDependencyCache.Put("foo", "bar3", "bar3")
```

Comment:
Can these tests be added as part of the table test?

Reply:
will do

Reply:
Its part of a single test that checks multiple values